src/haddock/libs/libplots.py

```diff
@@ -1433,33 +1453,70 @@ def fig_to_html(
 
     Parameters
     ----------
+    fig : Figure
+        A Figure object created by Plotly
+    fpath : Union[str, Path]
+        Where to write the content
     json_content : str
         plotly json content
-    
     plot_id : int
         plot id to be used in the html content
-    
     figure_height : int
         figure height (in pixels)
-    
     figure_width : int
         figure width (in pixels)
+    offline : bool
+        If set to False, use the cdn url to obtain the javascript content
+        for the rendering.
     """
     # Convert to json
     json_content = fig.to_json()
     # Create custom html file
     html_content = create_html(
         json_content,
         plot_id=plot_id,
+        plotly_js_import=offline_js_manager(fpath, offline),
         figure_height=figure_height,
         figure_width=figure_width,
-        offline=offline,
         )
     # Write it
     Path(fpath).write_text(html_content)
 
 
-def make_traceback_plot(tr_subset, plot_filename):
+def offline_js_manager(fpath: FilePath, offline: bool) -> str:
+    """Build string to access plotly javascript content.
+
+    Parameters
+    ----------
+    fpath : FilePath
+        Path to the figure about to be written.
+    offline : bool
+        Should this figure use the offline approache, or not.
+
+    Returns
+    -------
+    plotly_js_import : str
+        HTML solution for the importation of the plotly javascript content.
+    """
+    # Case where offline isrequired
+    if offline:
+        # Obtain directory where the figure should be written
+        fig_dir = Path(fpath).parent
```

Comment:
Lot of the examples have multiple caprieval modules. 

This will cause a plotly.js file in analysis/XX_caprieval, 

would be nice if the plotly.js could be shared between modules to reduce size even more.

Reply:
I totally agree with that.
Also other modules generating plots will have to use this file.

the issue is:
- I do not know how to access the relative path of it.
- for analysis directory, it would be `../../data/jsbundle/plotly.js`
- for other modules, it would be `../data/jsbundle/plotly.js`

I could not figure it out yet...

Reply:
When code is run the cwd is run_dir/XX_caprieval_analysis so the path on fs and in http is different see https://github.com/haddocking/haddock3/blob/d24a510d70727488465ce28f01a5bef3a649b92c/src/haddock/libs/libplots.py#L994-L995

Reply:
> I totally agree with that. Also other modules generating plots will have to use this file.
> 
> the issue is:
> 
> * I do not know how to access the relative path of it.
> * for analysis directory, it would be `../../data/jsbundle/plotly.js`
> * for other modules, it would be `../data/jsbundle/plotly.js`
> 
> I could not figure it out yet...

Why not something simple like this:

```python
def find_file(filename: str) -> Optional[Path]:
    search_paths = [Path('..').parent, Path('..')]
    for path in search_paths:
        full_path = path / filename
        if full_path.is_file():
            return full_path
    return None
```

Reply:
Its an idea, that must be fine tuned, but why not

Reply:
OK, tried an implementation following the suggestion idea